server-spi-private/src/main/java/org/keycloak/models/session/UserSessionPersisterProvider.java

```diff
@@ -78,6 +109,19 @@
     Stream<UserSessionModel> loadUserSessionsStream(Integer firstResult, Integer maxResults, boolean offline,
                                                     Integer lastCreatedOn, String lastUserSessionId);
 
+    /**
+     * Retrieves the count of user sessions
+     * @param offline
+     * @return
+     */
     int getUserSessionsCount(boolean offline);
 
+    /**
+     * Retrieves the count of user client-sessions for the given client
+     * @param offline
+     * @param clientModel
+     * @return
+     */
+    int getUserSessionsCount(boolean offline, ClientModel clientModel);
```

Comment:
I'll adapt and generally agree, but I think the realm makes no sense here since, the sole real implementation queries the `PersistentClientSessionEntity` via underlying query `PersistentClientSessionEntity.findClientSessionsCountByClient` filtered by the given client id. Also the entity `PersistentClientSessionEntity` does not have a `realmId` attribute.

I can add the parameter but I'd need to ignore it in the JPA implementation.

Reply:
Fixed, by adding the `RealmModel` parameter which is ignored by the JPA implementation.